channel.go

```diff
@@ -184,6 +229,60 @@ func (ch *channel) WriteFcall(ctx context.Context, fcall *Fcall) error {
 	return ch.bwr.Flush()
 }
 
+// maybeTruncate will truncate the message to fit into msize on the wire, if
+// possible. If the message cannot be truncated, an error will be returned and
+// the message should not be sent.
+//
+// A nil return value means the message can be sent without
+func (ch *channel) maybeTruncate(fcall *Fcall) error {
+	// If we are going to overflow the msize, we need to truncate the write to
+	// appropriate size or throw an error in all other conditions.
+	size := ch.msgmsize(fcall)
+	if size <= ch.msize {
+		return nil
+	}
+
+	// overflow the msize, including the channel message size fields.
+	overflow := size - ch.msize
+
+	// for certain message types, just remove the extra bytes from the data portion.
+	switch msg := fcall.Message.(type) {
+	// TODO(stevvooe): There are two more problematic message types:
+	//
+	// Tread: We can rewrite msg.Count so that a return message will be
+	// under msize. This is more defensive than anything but will ensure
+	// that calls don't fail on sloppy servers.
+	//
+	// Rread: while we can employ the same truncation fix as Twrite, we
+	// need to make it observable to upstream handlers. On the server side,
+	// this can likely be done in server.go or in dispatch by intercepting
+	// Tread.
+
+	case MessageTwrite:
+		if len(msg.Data) < overflow {
+			// paranoid: if msg.Data is not big enough to handle the
+			// overflow, we should get an overflow error. MSize would have
+			// to be way too small to be realistic.
+			break
+		}
+
+		// The truncation is reflected in the return message (Rwrite) by
+		// the server, so we don't need a return value or error condition
+		// to communicate it.
+		msg.Data = msg.Data[:len(msg.Data)-overflow]
```

Comment:
msg seems to be a copy of fcall.Message (I launched test from #31) on this branch with debugging.
adding `fcall.Message = msg` fixes the issue